pkg/cloudevents/transport/http/message.go

```diff
@@ -77,3 +81,32 @@ func (m Message) CloudEventsVersion() string {
 
 	return ""
 }
+
+func readAllClose(r io.ReadCloser) ([]byte, error) {
+	if r != nil {
+		defer r.Close()
+		return ioutil.ReadAll(r)
+	}
+	return nil, nil
+}
+
+// FromRequest initializes a Message from a http.Request
+func (m *Message) FromRequest(r *http.Request) (err error) {
```

Comment:
don't name the return

Reply:
Why? That has semantic meaning in Go and is necessary if you want to manipulate return values in a defer() statement. It can also be helpful as documentation if there are multiple same-type return values. 

None of that applies here, I'll change it (just means extra var statements in the body) but why the rule?